include/seqan3/std/charconv

```diff
@@ -0,0 +1,1010 @@
+{
+    // The locale issue:
+    // std::from_chars is documented to be locale independent. The accepted patterns
+    // are identical to the one used by strtod in the defailt ("C") locale.
+    //
+    // The functions strto[d/f/ld] used here are locale dependent but
+    // setting the locale manually by std::setlocale is not thread safe.
+    // So for the time being this workaround is locale dependent.
+
+    if (*first == '+') // + is permitted in function strto[d/f/ld] but not in from_chars
+        return {last, std::errc::invalid_argument};
+
+    float tmp{};
+    ptrdiff_t constexpr buffer_size = 10000;
+    char hex_buffer[buffer_size];
+    char * start;
+    char * end;
+
+    if (fmt != std::charconv::chars_format::general)
+    {
+        bool exponent_is_present{false};
+        for (auto it = first; it != last; ++it)
+        {
+            if (seqan3::is_char<'e'>(*it) || seqan3::is_char<'E'>(*it))
+            {
+                exponent_is_present = true;
+                break;
+            }
+        }
+
+        if (fmt == std::charconv::chars_format::scientific &&
+            !exponent_is_present)
+            return {last, std::errc::invalid_argument};
+
+        if (fmt == std::charconv::chars_format::fixed      &&
+            exponent_is_present)
+            return {last, std::errc::invalid_argument};
+    }
+
+    // If hex format is explicitly expected, the 0x prefix is not allowed in the
+    // the original sequence according to the std::from_chars cppreference
+    // documentation.
+    // In order to use strto[f/d/ld], the prefix must be prepended to achieve
+    // correct parsing. This will also automatically lead to an error if the
+    // original sequence did contain a 0x prefix and thus reflect the correct
+    // requirements of std::from_chars.
+    if (fmt == std::charconv::chars_format::hex)
+    {
+        hex_buffer[0] = '0';
+        hex_buffer[1] = 'x';
+        for (unsigned i = 0; i < std::min(buffer_size - 2, last - first); ++i)
+            hex_buffer[i+2] = first[i];
+
+        start = &hex_buffer[0];
+        end = &hex_buffer[0] + sizeof(hex_buffer);
+    }
+    else
+    {
+        start = const_cast<char *>(first);
+        end = const_cast<char *>(last);
+    }
+
+    if constexpr (std::Same<seqan3::remove_cvref_t<non_integral_type>, float>)
+    {
+        tmp = strtof(start, &end);
+    }
+    if constexpr (std::Same<seqan3::remove_cvref_t<non_integral_type>, double>)
+    {
+        tmp = strtod(start, &end);
+    }
+    if constexpr (std::Same<seqan3::remove_cvref_t<non_integral_type>, long double>)
+    {
+        tmp = strtold(start, &end);
+    }
+
+    if (errno == ERANGE)
+    {
+        return {last, std::errc::result_out_of_range};
+    }
+    else if (tmp == 0 && end == first)
+    {
+        return {last, std::errc::invalid_argument};
+    }
+
+    // Success.
+    value = tmp;
+    return {last, {}};
+}
+
+} // inline namespace charconv to trick the compiler
+
```

Comment:
Depending on how much of this file is copy'n'paste from LLVM, it might be easier to copy the entire file as-is,  save it as charconv_llvm and include based on some condition OR we import it properly, but then we should adapt the naming of the variables so they don't violate anything and also move stuff to detail that is detail.